src/main/java/cms/gov/madie/measure/utils/JsonUtil.java

```diff
@@ -632,4 +644,53 @@ public static String getTestCaseJson(String json) throws JsonProcessingException
     }
     return null;
   }
+
+  // going through JsonNode, converts any datetime string into UTC datetime
+  public static JsonNode replaceNestedDateTimeStringValue(JsonNode node) {
+    if (node.isObject()) {
+      ObjectNode objectNode = (ObjectNode) node;
+      objectNode
+          .fields()
+          .forEachRemaining(
+              entry -> {
+                String fieldName = entry.getKey();
+                JsonNode childNode = entry.getValue();
+                String currentValue = childNode.asText();
+                String newValue = getNewValue(currentValue);
+                if (childNode.isTextual()) {
+                  objectNode.put(fieldName, newValue);
+                } else {
+                  replaceNestedDateTimeStringValue(childNode);
+                }
+              });
+    } else if (node.isArray()) {
+      node.forEach(childNode -> replaceNestedDateTimeStringValue(childNode));
+    }
+    return node;
+  }
+
+  // converts a value into UTC datetime, if the the passed in value is a datetime
+  // otherwise, returns the original value
+  static String getNewValue(String value) {
+    String newValue = value;
+    if (value.length() > 19 && (PATTERN.matcher(value.substring(0, 19)).matches())) {
+      ZonedDateTime dateTime = null;
+      ZonedDateTime adjustedDateTime = null;
+      try {
+        dateTime = ZonedDateTime.parse(value, FORMATTER);
+        adjustedDateTime = dateTime.withZoneSameInstant(ZoneId.of("UTC"));
+        newValue = adjustedDateTime.format(FORMATTER).replace("Z", "+00:00");
+      } catch (DateTimeParseException e) {
+        try {
+          // for older data without milliseconds
+          dateTime = ZonedDateTime.parse(value, FORMATTER_2);
+          adjustedDateTime = dateTime.withZoneSameInstant(ZoneId.of("UTC"));
+          newValue = adjustedDateTime.format(FORMATTER_2).replace("Z", ".000+00:00");
```

Comment:
Here and even in try block, Why are we replacing it again ? 
dateTime.withZoneSameInstant(ZoneId.of("UTC")); shoulnd't withZoneSameInstant already converts anytimezone to UTC ? and a formatter should already handle it right ?

something like this ?
newValue = adjustedDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);

Reply:
@RohitKandimalla the only reason I'm replacing them is because in the card, it says we need to force the timezone to be zero, if I don't replace it, it will display as: 2025-03-27T08:44:43Z format.

Reply:
Yeah I understood that later, we do not want a "Z" 

That is when I looked up and found this, Do you think we can try and see if this works ? instead of manually updating the value.
newValue = adjustedDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);

Reply:
aah ok, Thank for trying nvm then

Reply:
@RohitKandimalla yeah it did not work, it will end up with Z

Reply:
idk what is wrong with having "Z" isn't "Z" represents UTC timezone ?

Reply:
@RohitKandimalla yes it does represent UTC, but I think the card says this format: 
![image](https://github.com/user-attachments/assets/7d681f85-6771-4eec-8c00-bd2dc04ff811)